packages/core/src/Executor.ts

```diff
@@ -191,6 +193,12 @@ export class Executor {
       if(this.memory.getNodeStatus(ancestor.id) !== 'COMPLETE') return;
     }
 
+    this.markNodeComplete(node);
+  }
+
+  private markNodeComplete(node: Node) {
```

Comment:
How can we take into account if registered onComplete hooks are async and should be awaited? There is no protection for that here?